packages/synapse-react-client/src/components/SynapseTable/usePrefetchTableData.ts

```diff
@@ -246,3 +258,38 @@ export function usePrefetchTableData(): { dataHasBeenPrefetched: boolean } {
       !isLoadingUserGroupData,
   }
 }
+
+/**
+ * Uses react-query to prefetch an entire page of row data.
+ *
+ * @returns A function that will prefetch the data for a given page number.
+ */
+export function usePrefetchTableRows(): (pageNumber: number) => Promise<void> {
```

Comment:
New hook to prefetch a RowSet.

Given more time, it would be great to update the `usePrefetchResourcesInTable` to be imperative rather than a hook, then we could just prefetch the batches of entity headers, user IDs, file handles, and restriction information after we fetch the rowSet here. Then opening the next page of data would be super quick.

Reply:
Do you want to capture this performance optimization idea in another jira ticket?

Reply:
Sure

Reply:
https://sagebionetworks.jira.com/browse/PORTALS-3205